shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/controler/DatasetProcessingApi.java

```diff
@@ -120,4 +122,34 @@ ResponseEntity<Void> updateDatasetProcessing(
 			@Parameter(description = "dataset processing to update", required = true) @Valid @RequestBody DatasetProcessing datasetProcessing, BindingResult result)
 			throws RestServiceException;
 
+	@Operation(summary = "massiveDownloadDatasetsByProcessingIds", description = "If exists, returns a zip file of the inputs/outputs per processing corresponding to the given processing IDs")
+	@ApiResponses(value = {
+			@ApiResponse(responseCode = "200", description = "zip file"),
+			@ApiResponse(responseCode = "401", description = "unauthorized"),
+			@ApiResponse(responseCode = "403", description = "forbidden"),
+			@ApiResponse(responseCode = "404", description = "no dataset found"),
+			@ApiResponse(responseCode = "500", description = "unexpected error") })
+	@GetMapping(value = "/massiveDownloadByProcessing")
+	@PreAuthorize("hasRole('ADMIN') or (hasAnyRole('EXPERT', 'USER') and @datasetSecurityService.HasRightOnEveryDatasetOfProcessings(#processingIds, 'CAN_DOWNLOAD'))")
+	void massiveDownloadByProcessingId(
+			@Parameter(description = "id of the processing", required=true) @Valid
+			@RequestParam(value = "processingIds") List<Long> processingIds,
```

Comment:
Hi @DuckflipXYZ, thank you for this good PR. I am not sure, if sending the list of processingIds as request param is a good idea, as limited by number of chars, a Post would be a better way to do so? same for the method below

Reply:
and please harmonize the method names in GetMapping with @Operation above